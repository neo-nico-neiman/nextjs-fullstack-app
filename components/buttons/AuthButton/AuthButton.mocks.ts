import { IAuthButton } from './AuthButton';

const base: IAuthButton = { 
    // add props
};

const alt: IAuthButton = {
    // add props
};

export const mockAuthButtonProps = {
    base,
    alt,
};