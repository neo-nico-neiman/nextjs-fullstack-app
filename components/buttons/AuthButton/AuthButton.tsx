import AuthContext from '@state/auth/AuthContext';
import { useContext } from 'react';

export interface IAuthButton extends React.ComponentPropsWithoutRef<'button'> {}

const AuthButton: React.FC<IAuthButton> = ({ className, ...buttonProps }) => {
  const { authenticated, logIn, logOut } = useContext(AuthContext);

  return (
    <button
      {...buttonProps}
      className={`${className} btn-primary`}
      onClick={authenticated ? logOut : logIn}
    >
      {authenticated ? 'Sign Out' : 'Sign In'}
    </button>
  );
};

export default AuthButton;
