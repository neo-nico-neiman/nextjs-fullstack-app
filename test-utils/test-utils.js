// test-utils.js
import Providers from '@state/providers';
import { render } from '@testing-library/react';

const customRender = (ui, options = {}) =>
  render(ui, { wrapper: Providers, ...options });

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render };
