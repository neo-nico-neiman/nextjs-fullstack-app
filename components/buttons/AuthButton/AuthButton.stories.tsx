import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';

import AuthButton from './AuthButton';
import { mockAuthButtonProps } from './AuthButton.mocks';


export default {
  title: 'Components/AuthButton',
  component: AuthButton,
} as ComponentMeta<typeof AuthButton>;

const Template: ComponentStory<typeof AuthButton> = (args) => <AuthButton {...args} />;

export const Example = Template.bind({});
Example.args = {...mockAuthButtonProps.base};
