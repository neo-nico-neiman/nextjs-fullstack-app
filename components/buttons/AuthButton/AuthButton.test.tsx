import { render, screen } from '@test-utils/test-utils';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';

import AuthButton from './AuthButton';

describe('AuthButton', () => {
  test('Should match the snapshot', () => {
    const { asFragment } = render(<AuthButton />);
    expect(asFragment).toMatchSnapshot();
  });

  test('Should have text LogIn when user is logOut', () => {
    render(<AuthButton />);
    expect(
      screen.getByRole('button', { name: /sign in/i })
    ).toBeInTheDocument();
  });

  test('Should toggle states', async () => {
    render(<AuthButton />);
    const signIn = screen.getByRole('button', { name: /sign in/i });
    expect(signIn).toBeInTheDocument();
    await userEvent.click(signIn);

    const signOut = screen.getByRole('button', { name: /sign out/i });
    expect(signOut).toBeInTheDocument();
    await userEvent.click(signOut);

    expect(signIn).toBeInTheDocument();
  });
});
