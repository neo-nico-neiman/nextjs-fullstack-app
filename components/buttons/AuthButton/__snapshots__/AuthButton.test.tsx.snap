// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthButton Should match the snapshot 1`] = `[Function]`;
